### ISSUE TITLE
Window lag after long idle time

### ISSUE THREAD

**Issue by runfalk** [linuxmint/cinnamon#3971]
I have an issue due to a recent update 2-4 weeks ago for Linux Mint 17.1. I run Cinnamon 2.4.6. It appears to happen after it has idled for an extended period of time, i.e. overnight. 

There is severe stuttering when moving windows, and scrolling in Firefox. Closing all open windows does not help. The only way to recover is logging out/rebooting or reloading Cinnamon with Alt+F2 and typing r.

I use the default open source AMD drivers. I have reproduced the issue on another machine which runs proprietary NVIDIA drivers. I normally leave Firefox and Skype running. I have not noticed abnormally high CPU usage or RAM.

**Comment by runfalk** [comment]
Further investigation suggests CPU usage does become elevated when moving windows while in this laggy state. When moving a windows a core instantly utilizes 100%. Memory usage seems OK, virtual 1490M and reserved 201M. Restarting Cinnamon brings memory usage down to 104M (reserved).

**Comment by 1TheBlueWii1** [comment]
I have been facing this issue with Mint 17.1 for sometime. I have 2 PCs that run Mint, an i5 and a Pentium 4 (s478), and both will start lagging after being used or remained idle for about an hour or longer. Restarting Cinnamon fixes the lag, but it happens again after another hour.

I recently learned an efficient way to removing the lag. Right-clicking on the panel, going to Troubleshooting... > Looking Glass > Memory and clicking on the 'Full Garbage Collection' button removes the lag without waiting for the desktop to restart.

Is there a way to run 'Full Garbage Collection' as a terminal command? I would like to make a simple script to run it every 5 minutes so I can avoid this completely.

**Comment by megz** [comment]
this happens for me since mint 17.1

the new updates were supposed to fix it and i actually thought they had for a while as i had noticed no stutter or lag whilst gaming for about an hour. but i actually just booted my machine and left it on the desktop for 1.20hrs with no activity and it was back with a vengeance.

troubleshoot --> restarting cinnamon fixes it temporarily. its a workaround but has been present for 3 months now.

My setup: Cinnamon 2.4.8 , Kernal happens on both 3.13 and my current 3.16.0-34-generic 

processor is AMD: GPU is Nvidia 7 series desktop GPU with latest beta 349 drivers.

everything else is standard, no tweaks no desklets, no applets, no extensions. no extra icons on the panel or desktop.

**Comment by mohammad-sn** [comment]
@1TheBlueWii1 I don't know any way to run "Full Garbage Collection" from terminal, but one can do it from within Cinnamon, with this command:
```javascript
const System = imports.system;
System.gc();
```
If you still need it, I can provide an applet to run these commands on applet click (or periodicly).

**Comment by runfalk** [comment]
I have not experienced this since updating to 2.4.8. I don't know if that was what fixed it or some other update.

**Comment by Vahan86** [comment]
@runfalk, is this still an issue for you? If not, can you please close it?

**Comment by runfalk** [comment]
I have since reporting changed my graphics card (Nvidia 970 GTX) and am now running the latest proprietary Nvidia drivers as well as Linux Mint 18, Cinnamon 3.0.7. I still have some window lag though it is less than it used to be.

I have noticed that the lag mainly appears when I have one window open that plays some kind of video (or gif). Closing that window will immediately make both browsing and moving windows a lot snappier.

To sum up: I still experience that windows are a lot jerkier than they would be on Windows or on my laptop that runs macOS. The issue seems to be related to windows that does some kind of animation in them (maybe redrawing).

**Comment by howdev** [comment]
this happens again with cinnamon 4. After long idle time every application is slow.  Restart cinnamon fixes the problem. Happens often, almost every time when I come back from long idle time, cinnamon is slow

**Comment by icarter09** [comment]
We believe this issue is fixed. Please test against the latest Cinnamon version (4.2). 

https://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/

**Comment by stz184** [comment]
I still experience this issue with the latest Linux Mint 20.

**Comment by icarter09** [comment]
@stz184 instead of reopening this ticket, could you create a new ticket addressing the issue? Thanks.

**Comment by stz184** [comment]
> @stz184 instead of reopening this ticket, could you create a new ticket addressing the issue? Thanks.

Yes, of course. The new issue is created here https://github.com/linuxmint/cinnamon/issues/9628

**Comment by howdev** [comment]
@stz184 Hello Vladmir, I was having the same issue with Nvidia driver, but seems gone when I was using  Cinnamon 4.2. Now I just upgraded to Mint 20 with Cinnamon 4.6 no problem there either.
